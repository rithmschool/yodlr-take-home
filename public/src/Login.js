import React, { useContext, useState } from 'react';
import { Form, FormGroup, Button, Input } from 'reactstrap';
import { CurrentUserContext } from './YodlrContext';
import { Redirect } from 'react-router-dom';

function Login ({ login }) {
    const currentUser = useContext(CurrentUserContext);
    const initialState = {id: "", password: ""};
    const [formData, setFormData] = useState(initialState);
    

    // handle user form input before submit
    function handleChange (evt) {
        const { name, value } = evt.target;
        setFormData(fData => ({
          ...fData,
          [name]: value
        }));
    }
    
    // handle form submission
    async function handleSubmit (evt) {
        evt.preventDefault();
        try{
        await login(formData);
        setFormData(initialState);
        } catch (err) {
            return( <p>Bad login, please try again.</p>)
        };
    }


    // display login form
    return (
        <div>
            <h2>Admin Login</h2>
            <Form onSubmit={handleSubmit}>
                <FormGroup>
                    <Input type="text" name="id" placeholder="ID" onChange={handleChange} required/>
                </FormGroup>
                <FormGroup>
                    <Input type="text" name="password" placeholder="Password" onChange={handleChange} required/>
                </FormGroup>
                <Button>Submit</Button>
            </Form>
        </div>
    )
}

export default Login;