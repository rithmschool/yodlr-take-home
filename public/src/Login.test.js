import React from "react";
import { render } from "@testing-library/react";
import Login from './Login';

it("renders without crashing", function() {
  render(<Login />);
});

it("matches snapshot", function() {
  const {asFragment} = render(<Login />);
  expect(asFragment()).toMatchSnapshot();
});