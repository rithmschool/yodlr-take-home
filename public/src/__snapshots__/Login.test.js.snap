// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Admin Login
    </h2>
    <form
      class=""
    >
      <div
        class="form-group"
      >
        <input
          class="form-control"
          name="id"
          placeholder="ID"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <input
          class="form-control"
          name="password"
          placeholder="Password"
          required=""
          type="text"
        />
      </div>
      <button
        class="btn btn-secondary"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
